ateam_kenobi/src/play_helpers/easy_move_to.cpp

```diff
@@ -171,4 +183,72 @@ void EasyMoveTo::drawTrajectoryOverlay(
   }
 }
 
+std::optional<ateam_msgs::msg::RobotMotionCommand> EasyMoveTo::generateEscapeVelocity(
+  const World & world,
+  const Robot & robot,
+  std::vector<ateam_geometry::AnyShape> obstacles)
+{
+  if (planner_options_.use_default_obstacles) {
+    path_planning::AddDefaultObstacles(world, obstacles);
+  }
+  path_planning::AddRobotObstacles(world.our_robots, robot.id, obstacles);
+  path_planning::AddRobotObstacles(world.their_robots, obstacles);
+  const auto robot_footprint = ateam_geometry::makeDisk(
+    robot.pos,
+    kRobotRadius + planner_options_.footprint_inflation);
+  const auto colliding_obstacle = path_planning::GetCollidingObstacle(robot_footprint, obstacles);
+  if (!colliding_obstacle) {
+    return std::nullopt;
+  }
+
+  const double safe_speed = 0.5;
+
+  auto msg_from_vector = [](const ateam_geometry::Vector & v) {
+      ateam_msgs::msg::RobotMotionCommand command;
+      command.twist.linear.x = v.x();
+      command.twist.linear.y = v.y();
+      return command;
+    };
+
+  if (std::holds_alternative<ateam_geometry::Circle>(*colliding_obstacle)) {
```

Comment:
Have we hit any edge cases where we are close enough to hit multiple obstacles? In those cases what happens?

Reply:
We almost certainly have hit that before. The code as is will pick the first one in the obstacles list. In practice, the order the obstacles are checked in is:
1. default obstacles (ie defender boxes)
2. our robots (lowest ID first)
3. their robots (lowet ID first.
The escape velocity will be generated from just that one obstacle, which may or may not be a useful velocity depending on the arrangement of the obstacles.
It's probably not worth fixing that behavior now, but definitely worth adding a TODO to remind us that this limitation exists.